python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
I tested with pyspark, and found removing redundant code is safe.

But I failed to test 2.10 repl? When I build with -Pscala-2.10, it failed with below error:
```
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-versions) @ spark-parent_2.11 ---
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.BannedDependencies failed with message:
Found Banned Dependency: org.scala-lang.modules:scala-xml_2.11:jar:1.0.2
Found Banned Dependency: org.scalatest:scalatest_2.11:jar:2.2.6
Use 'mvn dependency:tree' to locate the source of the banned dependencies.

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce (enforce-versions) on project spark-parent_2.11: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
```

I found even with -Pscala-2.10, sub-module still depends on "spark-parent_2.11".
Any suggestions?